File: 46218923-main/week6/pset6/sentimental-readability/readability.py
# TODO
from cs50 import get_string
import re


input = get_string("Text: ")
i = input.lower()

# found regex (re) here: https://stackoverflow.com/a/73209272
count_letters = len(re.findall(r"[a-z]", i))
# code testing
# print(count_letters)

# studying this tutorial: https://datagy.io/python-count-words/
count_words = len(i.split())
# code testing
# print(count_words)

# similar approach as for letters count
count_sentences = len(re.findall(r".+?[\.|!|?]+", i))
# code testing
# print(f"Sentences: {count_sentences}")
# average number of letters per 100 words in the text
L = (count_letters / count_words) * 100

# average number of sentencs per 100 words in the text
S = (count_sentences / count_words) * 100

X = 0.0588 * L - 0.296 * S - 15.8

if X < 1:
    print("Before Grade 1")

elif X >= 16:
    print("Grade 16+")

else:
    print(f"Grade {round(X)}")

File: CS50_Q42023/week2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");
    // printf("Text: %s\n", text);
    // printf("%i letters\n", count_letters(text));
    // printf("%i words\n", count_words(text));
    // printf("%i sentences\n", count_sentences(text));

    float L = (float) count_letters(text) / (float) count_words(text) * 100;
    float S = (float) count_sentences(text) / (float) count_words(text) * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

// function count letters
int count_letters(string text)
{
    int letters = 0;

    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    return letters;
}

// function count words
int count_words(string text)
{
    int words = 1;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isspace(text[i]))
        {
            words++;
        }
    }
    return words;
}

// function count sentences
int count_sentences(string text)
{
    int sentences = 0;
    int i = 0;

    for (i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    return sentences;
}
